### ISSUE TITLE
您好，我正在跑这个代码，但是因为受到组内资源的限制，我们只有两张3090的显卡，我将学习率不变并将epoch增大为50个，但是在dota-v1.0上的数据结果只有46.6，训练时长为3个多小时，我不知道是不是应该调整什么参数（学习率尝试过调整为一半或者是之前的四分之一，但是在前50个epoch的精度都没有超过50），期盼您的解答。因为我看您设置的epoch只有12.

### ISSUE THREAD

**Issue by Joan-L1019** [HVision-NKU/Strip-R-CNN#5]
[20250222_123948.log](https://github.com/user-attachments/files/18921919/20250222_123948.log)

**Comment by YXB-NKU** [comment]
Thank you for your interest in our work!

One important point to emphasize is that even if you are using only one GPU card, the number of epochs should be set to 12. Training for too many epochs may lead to overfitting, which can result in decreased performance. 

Additionally, if you are using two 3090 GPUs, you can set the batch size to 2 on each gpu card, making the total batch size 4. 

You can first try keeping the learning rate at 0.0001 (some people have asked me about this before, and it seems that keeping the learning rate unchanged worked better in their experiments). 

If the results are not satisfactory, you can try halving the learning rate, i.e., setting the learning rate (lr) to 0.00005. There is no need to adjust any other parameters.

Another thing : 
you should modify the dotav1.py file as shown below :
    train=dict(
        type='DOTADataset',
        ann_file='/home/cipan/smallobject/DOTA/**trainval**/annfiles/',
        img_prefix='/home/cipan/smallobject/DOTA/**trainval**/images/',

**Comment by Joan-L1019** [comment]
谢谢您的回复，我在数据集文件里的名字和dotav1.py里面的名字里面是统一的都是train，这个应该没有什么问题，然后我将Batch_size调整为之前的两倍进行了实验，但是训练的结果和我上传的前一个log文件没有太大差距，所以我仍然不知道是哪里的问题，麻烦您再帮我看看。因为原始标签前两行是冗余信息，所以我在dota.py文件读取的时候使用两个f.readline()进行跳过，这个修改是必要的，否则标签不对应跑不起来。别的都是原始内容，我没有修改过。

![Image](https://github.com/user-attachments/assets/d38b9ab7-20e7-4d6c-9f43-7d3e3402e167)

[20250222_191548.log](https://github.com/user-attachments/files/18922375/20250222_191548.log)
[P0000.txt](https://github.com/user-attachments/files/18922376/P0000.txt)

**Comment by Joan-L1019** [comment]
[20250222_191548.log](https://github.com/user-attachments/files/18922411/20250222_191548.log)

**Comment by YXB-NKU** [comment]
yes, I know that the name of your dataset is correct, but during training, you need to use the **trainval** set instead of the **train** dataset.

**Comment by Joan-L1019** [comment]
您好，非常感谢您的帮助，我对数据集进行了处理（将train+val=trainval）使用了40分钟重新进行了训练，精度提升到了49.8，但是仍然没有突破50大关，还是不太清楚到底是怎么回事，感觉就算将lr降为之前的一半，也只会提升一点点，不知道是不是我哪里做错了

[20250222_201128.log](https://github.com/user-attachments/files/18922795/20250222_201128.log)

**Comment by YXB-NKU** [comment]
After carefully examining the log file, I found that there is an issue with the size of your dataset. In the log, it shows **Epoch [2][50/467]**, but when the batch size is 4, the number 467 should actually be around 3200. I suspect the reason is that you haven't performed the **split** processing on the DOTA dataset.

How to split the dataset of DOTA？  Here are the detailed steps：
https://github.com/open-mmlab/mmrotate/blob/main/tools/data/dota/README.md

**Comment by Joan-L1019** [comment]
非常感谢您高效且有效的帮助，这个是我最新训练的结果，我使用您上述提及的工具进行了切图，但是看训练结果似乎还没有收敛，我是否应该再延长epoch呢？或者修改其他的参数，最后一轮的mAP:66.0。期盼您的回复。

![Image](https://github.com/user-attachments/assets/754d2d14-f8d1-4d03-b4a8-8e7634eb3c13)

[20250222_223022.log](https://github.com/user-attachments/files/18928021/20250222_223022.log)

**Comment by YXB-NKU** [comment]
maybe there is some bug , i can not find your issue here.

![Image](https://github.com/user-attachments/assets/8ba8067c-d56c-44d3-bf34-9998a01f33ef)

about this question ,  I guess you're still encountering some issues while handling the dataset. When you use two GPUs for training and the total batch size is 4, in the log `Epoch [4][50/2191]`, the number `2191` should actually be `3200` (which is a fixed value). So, you might want to double-check if there's an issue with your  configuration file of dataset processing, and also verify whether both the training and validation datasets are being used correctly.

**Comment by Joan-L1019** [comment]
您好非常感谢您的帮助，我已经完全实现了您的代码，没有任何问题，我出错的原因是，之前在没有使用分割标签以前（因为原始标签前两行是冗余信息，所以我在dota.py文件读取的时候使用两个f.readline()进行跳过，这个修改是必要的，否则标签不对应跑不起来。），但是在分割标签之后忘记将这两行代码删除，您的指导非常细致有效。祝您天天好心情，次次中顶会！！

**Comment by Joan-L1019** [comment]
您好，我使用了最新的代码对dota数据集进行运行没有任何问题，但是在运行dior数据集的时候出现一个问题，就是在对pth进行保存之后，会出现10个map的推理结果，而且这10个结果各不相同，但是也不太清楚是因为什么，log附在后面，辛苦您百忙之中帮我看看。期待您的回信。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "HVision-NKU/Strip-R-CNN"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年2月23日(星期天) 下午2:58
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [HVision-NKU/Strip-R-CNN] 您好，我正在跑这个代码，但是因为受到组内资源的限制，我们只有两张3090的显卡，我将学习率不变并将epoch增大为50个，但是在dota-v1.0上的数据结果只有46.6，训练时长为3个多小时，我不知道是不是应该调整什么参数（学习率尝试过调整为一半或者是之前的四�= 86之一，但是在前50个epoch的精度都没有超过50），期盼您的解答。因为我看您设置的epoch只有12. (Issue #5)



   
maybe there is some bug , i can not find your issue here.
 
image.png (view on web)
 
about this question ,  I guess you're still encountering some issues while handling the dataset. When you use two GPUs for training and the total batch size is 4, in the log Epoch [4][50/2191], the number 2191 should actually be 3200 (which is a fixed value). So, you might want to double-check if there's an issue with your  configuration file of dataset processing, and also verify whether both the training and validation datasets are being used correctly.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  YXB-NKU left a comment (HVision-NKU/Strip-R-CNN#5)
 
maybe there is some bug , i can not find your issue here.
 
image.png (view on web)
 
about this question ,  I guess you're still encountering some issues while handling the dataset. When you use two GPUs for training and the total batch size is 4, in the log Epoch [4][50/2191], the number 2191 should actually be 3200 (which is a fixed value). So, you might want to double-check if there's an issue with your  configuration file of dataset processing, and also verify whether both the training and validation datasets are being used correctly.

**Comment by YXB-NKU** [comment]
hello, I don't seem to see the log file. Could you please re-upload it?

**Comment by cripsgreen** [comment]
Hello, I followed your command line (python tools/data/dota/split/img_split.py --base-json \
tools/data/dota/split/split_configs/ss_trainval.json) to generate a new dataset. Then, I re-divided the dataset according to the official DOTA dataset’s train and val splits—meaning that what was previously in the training set remains in the training set, and the validation set remains unchanged. After training, the final result was around 0.47. Below is the training log. Could you take a look and see where the issue might be? @YXB-NKU [

[20250324_225918.log](https://github.com/user-attachments/files/19440475/20250324_225918.log)

](url)

**Comment by YXB-NKU** [comment]
First, check your `batch size` (bs) and `learning rate` (lr). When the batch size is 8, the learning rate should be 0.0001. If you adjust the batch size, the learning rate also needs to be adjusted proportionally. For example, if the batch size is increased to 16, you need to set the learning rate to 0.0002. Secondly, I noticed that the checkpoint doesn't seem to be pre-trained on ImageNet—it seems like you might have downloaded the wrong one.

![Image](https://github.com/user-attachments/assets/78141b93-be9b-4d7c-bfbf-8ffc27e9a988)
下载第四个权重，链接是https://drive.google.com/file/d/1_c2aXANKHl0cIBb370LNIkCyDmQpA3_o/view?usp=drive_link

**Comment by cripsgreen** [comment]
Thank you very much for your response! I have changed the pre-trained weights and adjusted the learning rate. Although the mAP has improved significantly, it still hasn't reached your reported result of around 0.82. I'm wondering if the issue could be related to how I split the dataset. Below is the training log. @YXB-NKU [

[20250325_224843.log](https://github.com/user-attachments/files/19460166/20250325_224843.log)

](url)

**Comment by YXB-NKU** [comment]
It seems that your training dataset has a relatively small number of images. Follow the steps here strictly to split the dataset, and put the trainval set together for training.

https://github.com/open-mmlab/mmrotate/blob/main/tools/data/dota/split/split_configs/ss_trainval.json

**Comment by cripsgreen** [comment]
If the training set and validation set are combined, what will be used as the validation set? @YXB-NKU

**Comment by YXB-NKU** [comment]
use the training set and validation set for training and validation set for val, this is a common setting.

**Comment by cripsgreen** [comment]
Thank you very much for your patient response. After running your code, the result was that the training set and validation set were combined. I just re-divided them according to the official DOTA v1 training and validation sets-meaning that what was previously in the training set remains in the training set, and the validation set remains unchanged.After the split, the training set has 15,749 samples, the validation set has 5,297 samples, totaling 21,046.

**Comment by Bhavana431** [comment]
Hello @cripsgreen,

Could you please share the dataset you used in your training? It would be very helpful for me.
